### PR TITLE
Improve typehints

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is not BC and can only be applied to master.

Part of #1935

### CODE REVIEW

File: src/Block/MediaBlockService.php [hunk 1]
```diff
@@ -184,7 +184,9 @@ public function validate(ErrorElement $errorElement, BlockInterface $block): voi
     }
 
     /**
-     * @return array<array-key, array-key>
```
Comment (VincentLanglet) [inline src/Block/MediaBlockService.php:187]: I would say that `@return array<int|string>` will avoid having two annotations.

File: src/Controller/Api/GalleryController.php [hunk 2]
```diff
@@ -245,7 +245,9 @@ public function getGalleryMediasAction($id): array
      *
      * @param int|string $id Gallery identifier
      *
-     * @return Collection<array-key, GalleryItemInterface>
+     * @return Collection<int|string, GalleryItemInterface>
+     *
+     * @phpstan-return Collection<array-key, GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Controller/Api/GalleryController.php:250]: What do we win with this over the `@return` ?

File: src/DependencyInjection/SonataMediaExtension.php [hunk 3]
```diff
@@ -395,8 +395,10 @@ public function prepend(ContainerBuilder $container): void
     /**
      * Checks if the classification of media is enabled.
      *
-     * @param array<string, class-string> $bundles
-     * @param array<string, mixed>        $config
+     * @param array<string, string> $bundles
+     * @param array<string, mixed>  $config
+     *
+     * @phpstan-param array<string, class-string> $bundles
```
Comment (VincentLanglet) [inline src/DependencyInjection/SonataMediaExtension.php:401]: For the record, class-string is now suported by PHPStorm.

Don't know if it's worth moving it to `@phpstan-param` then.

File: src/DependencyInjection/SonataMediaExtension.php [hunk 4]
```diff
@@ -423,8 +425,10 @@ private function configureResizers(ContainerBuilder $container, array $config):
     }
 
     /**
-     * @param array<string, class-string> $bundles
-     * @param array<string, mixed>        $config
+     * @param array<string, string> $bundles
+     * @param array<string, mixed>  $config
+     *
+     * @phpstan-param array<string, class-string> $bundles
```
Comment (VincentLanglet) [inline src/DependencyInjection/SonataMediaExtension.php:431]: Same

File: src/Filesystem/Replicate.php [hunk 5]
```diff
@@ -185,8 +183,10 @@ public function getMetadata($key)
      * Gets the class names as an array for both adapters.
      *
      * @return string[]
+     *
+     * @phpstan-return class-string[]
```
Comment (VincentLanglet) [inline src/Filesystem/Replicate.php:187]: Might be used in the `@return`.

File: src/Model/Gallery.php [hunk 6]
```diff
@@ -50,7 +50,9 @@ abstract class Gallery implements GalleryInterface
     protected $defaultFormat = MediaProviderInterface::FORMAT_REFERENCE;
 
     /**
-     * @var Collection<array-key, GalleryItemInterface>
+     * @var Collection<int|string, GalleryItemInterface>
+     *
+     * @phpstan-var Collection<array-key, GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/Gallery.php:55]: Same

File: src/Model/GalleryInterface.php [hunk 7]
```diff
@@ -49,12 +49,16 @@ public function setDefaultFormat(?string $defaultFormat): void;
     public function getDefaultFormat(): ?string;
 
     /**
-     * @param Collection<array-key, GalleryItemInterface> $galleryItems
+     * @param Collection<int|string, GalleryItemInterface> $galleryItems
+     *
+     * @phpstan-param Collection<array-key, GalleryItemInterface> $galleryItems
```
Comment (VincentLanglet) [inline src/Model/GalleryInterface.php:54]: Same

File: src/Model/GalleryInterface.php [hunk 8]
```diff
@@ -49,12 +49,16 @@ public function setDefaultFormat(?string $defaultFormat): void;
...
      */
     public function setGalleryItems(Collection $galleryItems): void;
 
     /**
-     * @return Collection<array-key, GalleryItemInterface>
+     * @return Collection<int|string, GalleryItemInterface>
+     *
+     * @phpstan-return Collection<array-key, GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/GalleryInterface.php:61]: Same

File: src/Model/Media.php [hunk 9]
```diff
@@ -135,7 +135,9 @@ abstract class Media implements MediaInterface
     protected $size;
 
     /**
-     * @var Collection<array-key, GalleryItemInterface>
+     * @var Collection<int|string, GalleryItemInterface>
+     *
+     * @phpstan-var Collection<array-key, GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/Media.php:140]: Same

File: src/Model/MediaInterface.php [hunk 10]
```diff
@@ -135,12 +135,16 @@ public function getCategory(): ?object;
     public function setCategory(?object $category = null): void;
 
     /**
-     * @param Collection<array-key, GalleryItemInterface> $galleryItems
+     * @param Collection<int|string, GalleryItemInterface> $galleryItems
+     *
+     * @phpstan-param Collection<array-key, GalleryItemInterface> $galleryItems
```
Comment (VincentLanglet) [inline src/Model/MediaInterface.php:140]: Same

File: src/Model/MediaInterface.php [hunk 11]
```diff
@@ -135,12 +135,16 @@ public function getCategory(): ?object;
...
      */
     public function setGalleryItems(Collection $galleryItems): void;
 
     /**
-     * @return Collection<array-key, GalleryItemInterface>
+     * @return Collection<int|string, GalleryItemInterface>
+     *
+     * @phpstan-return Collection<array-key, GalleryItemInterface>
```
Comment (VincentLanglet) [inline src/Model/MediaInterface.php:147]: Same

File: src/DependencyInjection/SonataMediaExtension.php (hunk 3 above)
Comment (jordisala1991) [reply src/DependencyInjection/SonataMediaExtension.php:401]: I guess nothing it is just an alias right?

File: src/Filesystem/Replicate.php (hunk 5 above)
Comment (jordisala1991) [reply src/Filesystem/Replicate.php:187]: Not sure, vscode does not detect it, are you using it on admin bundle without the prefix?

Comment (VincentLanglet) [reply src/Filesystem/Replicate.php:187]: I found one `@param array<class-string, string[]> $adminClasses` and the rest is in the phpstan-return. So let's continue this way indeed. I'm too much phpstorm-oriented.

File: src/DependencyInjection/SonataMediaExtension.php (hunk 3 above)
Comment (VincentLanglet) [reply src/DependencyInjection/SonataMediaExtension.php:401]: I assume you were talking bout `array-key`.

Just ask on slack and
- `array<array-key, Foo>` is the same than `array<Foo>` 
- array-key is for int|string what mixed is for int|string|float|bool|... ; best would be to precise the keys.

In the case of
https://github.com/sonata-project/SonataMediaBundle/pull/2017/files#diff-ff8944e4b35328a08ac08926ea1feb404a7261265fa7bbd12fa923154613589dR191
looking at getFormatNamesByContext
https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Provider/Pool.php#L181
we can precise the signature to `array<string, string>`.

In the case of 
https://github.com/sonata-project/SonataMediaBundle/pull/2017/files#diff-280f9f70126ae1825f3748a0096963b6d600ebc5ca38b8ffe532214ebc654eaaR252
the array-key of the collection https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Model/Gallery.php#L53 could be changed to `int`
and so on

File: src/Block/MediaBlockService.php (hunk 1 above)
Comment (jordisala1991) [reply src/Block/MediaBlockService.php:187]: Fixed

File: src/DependencyInjection/SonataMediaExtension.php (hunk 3 above)
Comment (jordisala1991) [reply src/DependencyInjection/SonataMediaExtension.php:401]: Fixed all of them

File: src/Model/GalleryInterface.php (hunk 7 above)
Comment (jordisala1991) [reply src/Model/GalleryInterface.php]: Fixed

File: src/Model/GalleryInterface.php (hunk 8 above)
Comment (jordisala1991) [reply src/Model/GalleryInterface.php]: Fixed

File: src/Model/Media.php (hunk 9 above)
Comment (jordisala1991) [reply src/Model/Media.php]: Fixed

File: src/Model/MediaInterface.php (hunk 10 above)
Comment (jordisala1991) [reply src/Model/MediaInterface.php]: Fixed

File: src/Model/MediaInterface.php (hunk 11 above)
Comment (jordisala1991) [reply src/Model/MediaInterface.php]: Fixed

File: src/Controller/Api/GalleryController.php (hunk 2 above)
Comment (jordisala1991) [reply src/Controller/Api/GalleryController.php]: Fixed

File: src/Model/Gallery.php (hunk 6 above)
Comment (jordisala1991) [reply src/Model/Gallery.php]: Fixed

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**